src/AppRoutes.jsx

```diff
@@ -32,14 +31,7 @@ const AppRoutes = () => {
         path="/"
         element={session.info.isLoggedIn ? <Navigate to={path} replace /> : <Home />}
       />
-      <Route
-        path="/signup"
-        element={
-          <SessionProvider>
-            <Signup />
-          </SessionProvider>
-        }
-      />
+      <Route path="/signup" element={<Signup />} />
```

Comment:
The signup now manages its own state entirely (it doesn't interact with the session) so we don't need to mask the session or force the user to log out.